test/src/basic/fileformats.cpp

```diff
@@ -697,6 +695,47 @@ void describeDOTwithClusters()
 				AssertThat(cluster->nodes.size(), Equals(4));
 			}
 		});
+
+		it("reads assignment statements", []() {
+			const ResourceFile* file = ResourceFile::get(
+				"fileformats/dot/valid/assignments");
+			std::stringstream is{file->data()};
```

Comment:
Since you don't need `file` again, you could also use
`std::stringstream is{ResourceFile::get("fileformats/dot/valid/assignments")->data()}` directly.

(Same below.)

Reply:
While we're at it, why don't we put the `stringstream` inside `readDOT` too? We could save on many unused lines and extra variables by doing this

```c++
AssertThat(GraphIO::readDOT(GA, G, std::stringstream{ResourceFile::get("fileformats/dot/valid/arrowtypes")->data()}), Equals(true));
```